### ISSUE TITLE
naming scheme for generating ResourceSync documents

### ISSUE THREAD

**Issue by markmatney** [resourcesync/py-resourcesync#8]
I've observed the following about the way this implementation names generated sitemaps (resourcelist, changelist, etc.) when `ResourceSync.execute()` is invoked (for a set of 50000 resources or less):
- With strategy 0 (`resourcelist`):
  - On first invocation, a file `resourcelist_0000.xml` is created.
  - Subsequent invocations modify that file and keep the same name.
- With strategy 1 (`new_changelist`):
  - On first invocation, a file `changelist_0000.xml` is created.
  - On subsequent invocations, a file `changelist-index.xml` is created, and points to a series of `changelist_xxxx.xml`  files.
- With strategy 2 (`inc_changelist`):
  - On first invocation, a file `changelist_0000.xml` is created.
  - Subsequent invocations modify that file and keep the same name.

I've not tested on collections with more than 50000 resources, but per the [spec](https://www.openarchives.org/rs/1.1/resourcesync#ResourceListIndex), I expect there'd be a `resourcelist-index.xml` generated that points to a number of resourcelists.

I am thinking that whenever there is only one resourcelist or changelist for a set of resources, then it should be named simply `resourcelist.xml` or `changelist.xml`, without a `_0000` suffix, on the basis that a suffix like that is misleading/unnecessary when it is the only such sitemap for a set of resources.

Thoughts?

**Comment by hariharshankar** [comment]
>I am thinking that whenever there is only one resourcelist or changelist for a set of resources, then it should be named simply `resourcelist.xml` or `changelist.xml`, without a `_0000` suffix, on the basis that a suffix like that is misleading/unnecessary when it is the only such sitemap for a set of resources.

The main reason for using this naming scheme is consistency. If we name the first file either `resourcelist.xml` or `changelist.xml`, then the subsequent files with name `_***.xml` will have no continuity with the first file and will cause confusion. The current naming scheme, I think, is very intuitive even for people with no knowledge of the protocol.

**Comment by markmatney** [comment]
I agree that that would be confusing; whenever there is more than one `<capability>.xml` for a set of resources, it would suffice (and be a good idea) to rename the first file to `<capability>_0000.xml`. However, that may never even be necessary, since it's possible that only one changelist may ever exist for a set of resources (by exclusively invoking strategy 2 for changelist generation). But, when I _do_ see a file named `changelist_0000.xml`, I am inclined to expect to also see `changelist_0001.xml`, for example. I think a good sanity check would be to just call the singular changelist `changelist.xml`.
